massa-hash/src/hash_xof.rs

```diff
@@ -0,0 +1,57 @@
+use std::ops::{BitXor, BitXorAssign};
+
+use crate::{Hash, XOF_HASH_SIZE_BYTES};
+
+/// Extended Hash
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct HashXof(pub [u8; XOF_HASH_SIZE_BYTES]);
+
+impl HashXof {
+    /// Truncate the extended hash to a regular hash
+    pub fn clip_to_massa_hash(&self) -> Hash {
+        Hash::from_bytes(&self.0[..32].try_into().unwrap())
+    }
+
+    /// Compute from raw data
+    pub fn compute_from(data: &[u8]) -> HashXof {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(data);
+        let mut hash = [0u8; XOF_HASH_SIZE_BYTES];
+        let mut output_reader = hasher.finalize_xof();
+        output_reader.fill(&mut hash);
+        HashXof(hash)
+    }
+
+    /// Compute from key and value
+    pub fn compute_from_kv(key: &[u8], value: &[u8]) -> HashXof {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(key);
```

Comment:
```
hasher.update(key.len().to_be_bytes());
hasher.update(key);
hasher.update(value);
```

So that that Hash("abc", "def") != Hash("ab", "cdef")

Reply:
That is a very good point.

Reply:
Changed.